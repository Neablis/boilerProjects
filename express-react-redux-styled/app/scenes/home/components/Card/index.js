// Libraries
import React, { PureComponent } from 'react';

import styled, { ThemeProvider } from 'styled-components'

// Theme
import theme from 'services/theme'

// Styles
const CardContainer = styled.section`
  /* Color the border and text with theme.main */
  color: ${props => props.theme.white};
  background-color: ${props => props.theme.backgroundColor};
  text-align: center;
  border-radius: 15px;
  min-height: 150px;
  min-width: 100px;
  display: inline-block;
  margin: 20px;
`;

const CardHeader = styled.div`
  background-color: ${props => props.theme.color};
  border-radius: 15px 15px 0 0;
  height: 30px;
  width: 100%;
`;

const CardBody = styled.div`
  margin: 10px;
`;


class Card extends PureComponent {
  render() {
    const {
      header,
      children
    } = this.props;

    return (
      <ThemeProvider theme={theme}>
        <CardContainer>
          {
            header && (
              <CardHeader>{this.props.header}</CardHeader>
            )
          }
          <CardBody>
            {
              children
            }
          </CardBody>
        </CardContainer>
      </ThemeProvider>
    );
  }
}

export default Card
