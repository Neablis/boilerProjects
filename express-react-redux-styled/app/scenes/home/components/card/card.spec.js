import React from 'react'

import Card from './index'

test('render a Card', () => {
    const wrapper = shallow(
        <Card></Card>
    );
    expect(wrapper).toMatchSnapshot();
});

test('render a Card with a header', () => {
    const wrapper = shallow(
        <Card header="Has header"></Card>
    );
    expect(wrapper).toMatchSnapshot();
});