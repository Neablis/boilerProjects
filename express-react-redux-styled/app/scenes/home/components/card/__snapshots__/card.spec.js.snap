// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a Card 1`] = `
<ThemeProvider
  theme={
    Object {
      "alternativeColor": "#C9EDDC",
      "backgroundColor": "#4C3B4D",
      "color": "#7CE577",
      "white": "#ffffff",
    }
  }
>
  <card__CardContainer>
    <card__CardBody />
  </card__CardContainer>
</ThemeProvider>
`;

exports[`render a Card with a header 1`] = `
<ThemeProvider
  theme={
    Object {
      "alternativeColor": "#C9EDDC",
      "backgroundColor": "#4C3B4D",
      "color": "#7CE577",
      "white": "#ffffff",
    }
  }
>
  <card__CardContainer>
    <card__CardHeader>
      Has header
    </card__CardHeader>
    <card__CardBody />
  </card__CardContainer>
</ThemeProvider>
`;
